ckanext/harvest/harvesters/base.py

```diff
@@ -288,9 +288,21 @@ def _create_or_update_package(self, package_dict, harvest_object,
 
             if self.config and self.config.get('clean_tags', False):
                 tags = package_dict.get('tags', [])
-                tags = [munge_tag(t) for t in tags if munge_tag(t) != '']
-                tags = list(set(tags))
-                package_dict['tags'] = tags
+                if tags:
+                    if not isinstance(tags[0], dict):
```

Comment:
Please do not check the type like this (and the `if tags:` is not necessary). Python uses the [EAFP style](http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html#eafp-vs-lbyl), so you should rather wrap this in a try/except and handle the exception, something like this:

```python
try:
    # assume it's in the package_show form
   tags = [munge_tag(t['name']) for t in tags if munge_tag(t['name']) != '']
   ...
except TypeError: # a TypeError is raised if `t` above is a string
   # REST format: 'tags' is a list of strings
   tags = [munge_tag(t) for t in tags if munge_tag(t) != '']
   ...

Reply:
ok